go/service/user.go

```diff
@@ -623,3 +623,33 @@ func (h *UserHandler) UserCard(ctx context.Context, arg keybase1.UserCardArg) (r
 	}
 	return libkb.UserCard(mctx, uid, arg.UseSession)
 }
+
+func (h *UserHandler) BlockUser(ctx context.Context, username string) (err error) {
+	mctx := libkb.NewMetaContext(ctx, h.G())
+	defer mctx.TraceTimed("UserHandler#BlockUser", func() error { return err })()
+	return h.blockUserHelper(mctx, username, false)
```

Comment:
this is a confusing API. you call the `block` thing and pass it `false` to do the blocking and `true` to undo it. '__'  it looks like it's because `user/block` accepts a parameter for `unblock` which seems like the thing that's actually weird. i'd propose keeping that as isolated as possible inside the `blockUserHelper` and changing `unblock` on this method to something like `wantBlocked`.

Reply:
It's not too late to change the server side. What's the least confusing combo?

Reply:
i'm a believer in minimizing the number of `negative`s in parameters, because there will always be a reason to add a negative on top of something later, and triple negatives should be avoided. if i had my druthers, i think i'd try to make this read more declarative. like, alice wants bob to be blocked, or alice wants bob not to be blocked. so either a desired state (enum: "blocked" / "not blocked") or a `true` is for `blocked` and `false` is for `not blocked`. not sure the juice is worth the squeeze on going back to the server. it's more important to me that whatever is weird is totally isolated in one function.